client/src/api/routes/users.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery, UseQueryResult } from "react-query";
+import { fetcher } from "api/fetcher";
+import { doEmailSignIn, doEmailSignUp, doGoogleSignIn, doSignOut } from "api/firebase/functions";
+import { Failure } from "typings";
+
+/**
+ * @description POST /users/signIn with email & password
+ * @param {string} email 
+ * @param {string} password 
+ * @returns {UseQueryResult<boolean>} true on success, throws if fails
+ */
+export const useSignInWithEmail = (email: string, password: string): UseQueryResult<boolean> => {
+    return useQuery('signInWithEmail', async () => {
+        await doEmailSignIn(email, password);
+        const { data, status } = await fetcher.post<Failure>('/users/signIn');
```

Comment:
Same as my other comment, I'm not sure on the behavior of this, but wouldn't typescript throw an error if it doesn't always return failure?